compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -74,14 +74,24 @@ class StringInterpolatorOpt extends MiniPhase {
         case SOrRawInterpolator(strs, elems) =>
           if (tree.symbol == defn.StringContext_raw) Some(strs, elems)
           else { // tree.symbol == defn.StringContextS
+            import dotty.tools.dotc.util.SourcePosition
+            var stringPosition: SourcePosition = null
             try {
               val escapedStrs = strs.map { str =>
-                val escapedValue = StringContext.processEscapes(str.const.stringValue)
-                cpy.Literal(str)(Constant(escapedValue))
+                stringPosition = str.sourcePos // TODO: + length of quote(s)? Why?
+                val escaped = StringContext.processEscapes(str.const.stringValue)
+                cpy.Literal(str)(Constant(escaped))
               }
               Some(escapedStrs, elems)
             } catch {
-              case _: StringContext.InvalidEscapeException => None
+              case iee: StringContext.InvalidEscapeException => {
+                ctx.error(iee.getMessage() + "\n", stringPosition) //should be positioned at str.po
```

Comment:
Before, this didn't emit an error at compile time, only at runtime.

I could use a hand in positioning this properly.

Reply:
> I could use a hand in positioning this properly.

Are you looking for `tree.sourcePos` or something more specific ?

Reply:
I'd like to narrow it to a position of the actual error within the sourcePos.

Also, the transcript looks off (with the position too far "left") here, but I'm not sure anymore whether it just *looks* wrong or *is* wrong.

Reply:
This proved a challenge, position of the first part of the interpolation was off, and in test output the second line of error output is indented differently than the first, giving the impression the caret is in a different position than it's really in, but I think I got it.